secure_message/repository/retriever.py

```diff
@@ -46,38 +52,95 @@ def retrieve_message_list(page, limit, user, ru_id=None, survey=None, cc=None, c
 
         try:
             t = db.session.query(SecureMessage.msg_id, func.max(Events.date_time)  # pylint:disable=no-member
-                                 .label('max_date'))\
+                                 .label('max_date')) \
                 .join(Events).join(Status).outerjoin(Actors) \
                 .filter(and_(*conditions)) \
                 .filter(and_(*status_conditions)) \
                 .filter(or_(Events.event == EventsApi.SENT.value, Events.event == EventsApi.DRAFT_SAVED.value)) \
                 .group_by(SecureMessage.msg_id).subquery('t')
 
             if descend:
-                result = SecureMessage.query\
-                    .filter(SecureMessage.msg_id == t.c.msg_id) \
-                    .order_by(t.c.max_date.desc()).paginate(page, limit, False)
-
+                order = t.c.max_date.desc()
             else:
-                result = SecureMessage.query \
-                    .filter(SecureMessage.msg_id == t.c.msg_id) \
-                    .order_by(t.c.max_date.asc()).paginate(page, limit, False)
+                order = t.c.max_date.asc()
+
+            result = SecureMessage.query \
+                .filter(SecureMessage.msg_id == t.c.msg_id) \
+                .order_by(order).paginate(page, limit, False)
 
         except Exception as e:
             logger.error('Error retrieving messages from database', error=e)
             raise InternalServerError(description="Error retrieving messages from database")
 
         return True, result
 
+    @staticmethod
```

Comment:
I'm really confused to why these methods return True, result?

The calling function does an if on the status which is always true?

Reply:
Good catch . Me too , changed.

Reply:
Actually tried to change as I agree it looks wrong . Get lots of failing tests . Raised a tech debt ticket . Plan on looping back to this